Added note on cache commands over a sudo built container (#164)
sylabs/singularity-userdocs: build_env.rst

{Singularity} will cache SIF container images generated from remote
sources, and any OCI/docker layers used to create them. The cache is
created at ``$HOME/.singularity/cache`` by default. The location of the

> cache can be changed using the ``SINGULARITY_CACHEDIR`` environment

Reviewer: Personally prefer 'setting' here. If someone searches for 'setting environment variable' then it's more likely to give them what they need to know.

Singularity 3.1 comes with new commands for cleaning and listing the cache image files generated.

.. note::

    Running the cache commands with sudo privilege will consider cache location as ``/root/.singularity/cache``. The default location for cache without sudo privilege is ``~/.singularity/cache``.

>     Make sure that if you build a container with sudo privilege, you will need to consider the sudo location from the cache, and not the default.

Reviewer 2: I would changes this up a bit:

```suggestion
    If you build a container with sudo privilege (like a definition file), you cache will be stored in your root directory (``/root/.singularity/cache``) and not your home directory.

```

>     For example, running the following command with sudo privilege (considering the sudo privilege location for cache ``/root/.singularity/cache``):

Reviewer 2: I would take this out of the `note` block, and change it like this:

```suggestion
For example, running a action (``exec``, ``pull``, ``run``, etc...), command with sudo (or as root), cache will be stored in ``/root/.singularity/cache``, _if the ``SINGULARITY_CACHEDIR`` is **not** set._
```

You can add to that, and changes as you see fit :+1:

    .. code-block:: none

        $ sudo singularity cache list
        NAME                   DATE CREATED           SIZE             TYPE
        ubuntu_latest.sif      2019-01-31 14:59:32    28.11 Mb         library

>         ubuntu_18.04.sif       2019-01-31 14:58:44    27.98 Mb         library

Reviewer 2: I would update the output for the cache list command: *Add the summary after the cache*

```
    $ sudo singularity cache list 
NAME                     DATE CREATED           SIZE             TYPE
ubuntu_latest.sif        2019-05-07 12:53:14    28.11 MB         library
ubuntu_18.04.sif         2019-05-07 12:53:24    27.98 MB         library

There 2 containers using: 56.09 MB, 0 oci blob file(s) using 0.00 kB of space.
Total space used: 56.09 MB
```

>     and then cleaning the cache without sudo privilege (``singularity cache clean -a``) will not work, since the default cache location is ``~/.singularity/cache``.

Reviewer 2: Again, I would remove the spaces before the text, and change it like:

```suggestion
If you set ``SINGULARITY_CACHEDIR`` to something like ``SINGULARITY_CACHEDIR=/tmp/image-cache``, then all cache will be stored in `/tmp/image-cache`. Remember, when you use sudo, to add `-E` flag to your singularity command, like ``sudo -E singularity build test.sif test.def``.
```

And/or add a link to the page where it talks about `SINGULARITY_CACHEDIR`.

*These are just my suggestions, you can change it as you see fit.*

>     In this case you would need to run the clean command with sudo privilege:

Reviewer 2: ```suggestion
For example cleaning you cache in ``/root/.singularity/cache``:
```

>     .. code-block:: none

Reviewer 2: ```suggestion
.. code-block:: none
```

>         $ sudo singularity cache clean -a

Reviewer 2: And also add a example when `SINGULARITY_CACHEDIR` is set:

```
    # And with 'SINGULARITY_CACHEDIR' set, use the '-E' flag for sudo:
    $ sudo -E singularity cache clean --all
```

        NAME                     DATE CREATED           SIZE             TYPE

        There 0 containers using: 0.00 kB, 0 oci blob file(s) using 0.00 kB of space.

>         Total space used: 0.00 kB

Reviewer 2: This looks like the output for `cache list`, not `cache clean`. I would just remove all the output for the `cache clean` command.
